_posts/2020-02-05-keycloak-z-angularem-i-keycloakiem.md

```diff
@@ -0,0 +1,427 @@
+---
+layout:    post
+title:     "Keycloak - uwierzytelnianie i autoryzacja użytkownika w aplikacji Angular/Spring Boot"
+published: true
+date:      2020-02-05 08:00:00 +0100
+author:    mhoja
+tags:
+    - angular
+    - spring boot
+    - keycloak
+    - keycloak-spring-boot-2-starter
+    - keycloak-spring-boot-2-adapter
+    - keycloak-angular
+    - uwierzytelnianie użytkownika
+---
+
+Jak wykorzystać serwer Keycloak do logowania w naszej aplikacji?  
+Omówimy to na przykładzie gotowego projektu, który umożliwia użytkownikowi zalogowanie się do aplikacji z poziomu przeglądarki internetowej.
+
+## Czym jest Keycloak?
+
+Keycloak to serwer uwierzytelniania i autoryzacji na licencji open-source. Może zostać podłączony do LDAP/AD lub uwierzytelniać użytkowników przy użyciu Google, Facebooka itd.
+Posiada również konsolę administracyjną, w której możemy łatwo skonfigurować chociażby uprawnienia użytkowników.
+
+Więcej informacji na stronie [www.keycloak.org](https://www.keycloak.org/).
+
+## Projekt demo
+
+Projekt aplikacji znajduję się na GitHubie - [link do repozytorium.](https://github.com/Michuu93/spring-angular-keycloak)  
+Składa się on z aplikacji backendowej napisanej w Spring Boot (Kotlin) oraz frontendowej napisanej w Angular 8.
+
+Aby przedstawić sposób zaimplementowania uwierzytelniania użytkownika za pomocą serwera Keycloak, zabezpieczymy endpointy na backendzie oraz osobno część frontendową.
+
+### Serwer Keycloak
+
+Wykorzystamy serwer Keycloak uruchomiony na Dockerze. Wykonując w głównym folderze repozytorium polecenie:
+
+```shell
+docker run --rm --name keycloak-server -p 8180:8080 \
+    -e KEYCLOAK_USER=admin \
+    -e KEYCLOAK_PASSWORD=admin \
+    -e KEYCLOAK_IMPORT=/realm/realm-export.json \
+    -v $(pwd)/realm/:/realm/ \
+    jboss/keycloak
+```
+
+uruchomimy serwer Keycloak na porcie `8180`.
+
+Konsola administracyjna będzie dostępna pod adresem: [127.0.0.1:8180/auth/admin](http://127.0.0.1:8180/auth/admin/)  
+Zalogujemy się do niej przy użyciu loginu i hasła zdefiniowanego w `KEYCLOAK_USER` oraz `KEYCLOAK_PASSWORD`, w naszym przypadku `admin:admin`.
+
+#### Konfiguracja serwera
+
+Nie będziemy tutaj omawiać konfiguracji samego serwera Keycloak, ponieważ nie tego dotyczy wpis.
```

Comment:
Proponuję:
Sama konfiguracja serwera Keyclock, to temat na osobny wpis, a tutaj skupimy się na (...)